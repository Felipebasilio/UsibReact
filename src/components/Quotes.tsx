import React from 'react';
import { NavLink } from 'react-router-dom';
import ListQuotes from './ListQuotes'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus } from '@fortawesome/free-solid-svg-icons'

const Quotes = () => {
    return (
        <>
            <div className="container">
              <div className="my-5">
                <h2 className="col-12">Lista de Orçamentos</h2>
              </div>
              <p className="d-flex justify-content-end align-items-center" id="addQuote">
                Fazer novo orçamento
                <NavLink to="/addQuote" className="px-4 btn" activeClassName="active">
                  <FontAwesomeIcon icon={faPlus} />
                </NavLink>
              </p>
              <div>
                <table id="quotesTable" className="table text-center">
                  <thead>
                    <tr>
                      <th scope="col">#</th>
                      <th scope="col">Nome da empresa</th>
                      <th scope="col">Data de orçamento</th>
                      <th scope="col">Endereço</th>
                      <th scope="col">Email</th>
                      <th scope="col">Produto</th>
                      <th scope="col">Quantidade</th>
                      <th scope="col">Editar</th>
                      <th scope="col">Deletar</th>
                    </tr>
                  </thead>
                  <tbody>
                    <ListQuotes />  
                  </tbody>
                </table>
              </div>
            </div>
        </>
    );
  };
  
  export default Quotes;