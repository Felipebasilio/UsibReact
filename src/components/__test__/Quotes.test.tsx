import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';

import renderer from 'react-test-renderer'
import Quotes from '../Quotes';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

it('renders Quotes without crashing', () => {
    let div = document.createElement('div');
    ReactDOM.render( 
        <BrowserRouter>
            <Quotes />
        </BrowserRouter>
    , div );
    ReactDOM.unmountComponentAtNode(div);
});

it('Matches snapshot', () => {
    const QuotesComponent = renderer.create(
        <BrowserRouter>
            <Quotes />
        </BrowserRouter>
    ).toJSON();
    expect(QuotesComponent).toMatchSnapshot();
});


