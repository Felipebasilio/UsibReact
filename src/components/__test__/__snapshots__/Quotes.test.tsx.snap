// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="my-5"
  >
    <h2
      className="col-12"
    >
      Lista de Orçamentos
    </h2>
  </div>
  <p
    className="d-flex justify-content-end align-items-center"
    id="addQuote"
  >
    Fazer novo orçamento
    <a
      aria-current={null}
      className="px-4 btn"
      href="/addQuote"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </p>
  <div>
    <table
      className="table text-center"
      id="quotesTable"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            #
          </th>
          <th
            scope="col"
          >
            Nome da empresa
          </th>
          <th
            scope="col"
          >
            Data de orçamento
          </th>
          <th
            scope="col"
          >
            Endereço
          </th>
          <th
            scope="col"
          >
            Email
          </th>
          <th
            scope="col"
          >
            Produto
          </th>
          <th
            scope="col"
          >
            Quantidade
          </th>
          <th
            scope="col"
          >
            Editar
          </th>
          <th
            scope="col"
          >
            Deletar
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
